### PR TITLE
2022-11 CWG Motion 12

### LINKED ISSUE #5967: [2022-11 CWG Motion 12] P2718R0 Wording for P2644R1 Fix for Range-based for Loop

### PR DESCRIPTION
**Pull request by tkoeppe**
P2718R0 Wording for P2644R1 Fix for Range-based for Loop

Fixes #5967.
Fixes cplusplus/papers#1316
Fixes cplusplus/nbballot#471

### CODE REVIEW

**Comment by tkoeppe** [comment]
The "Annex-C style" for examples is described in https://github.com/cplusplus/draft/wiki/Specification-Style-Guidelines#formatting-and-structure.

File: source/statements.tex [hunk 1]
```diff
@@ -694,14 +694,30 @@
...
 \end{itemize}
-
 \begin{example}
 \begin{codeblock}
 int array[5] = { 1, 2, 3, 4, 5 };
 for (int& x : array)
   x *= 2;
 \end{codeblock}
 \end{example}
+\begin{note}
```
Comment (JohelEGP) [inline source/statements.tex:704]: The paper instructs
> In subclause 8.6.5 [stmt.ranged] add before Example 1:

This was added _after_ Example 1.

File: source/compatibility.tex [hunk 2]
```diff
@@ -69,6 +69,27 @@
                         // now equivalent to \tcode{arr.operator[](1, 2)} or ill-formed
 \end{codeblock}
 
+\rSec2[diff.cpp20.stmt]{\ref{stmt.stmt}: statements}
+
+\diffref{stmt.ranged}
+\change
+The lifetime of temporary objects in the \grammarterm{for-range-initializer}
+is extended until the end of the loop\iref{class.temporary}.
+\effect
```
Comment (JohelEGP) [inline source/compatibility.tex:78]: `\rationale` is missing.

File: source/statements.tex (hunk 1 above)
Comment (tkoeppe) [reply source/statements.tex:704]: See commit message.

File: source/compatibility.tex (hunk 2 above)
Comment (tkoeppe) [reply source/compatibility.tex:80]: Thanks, fixed!

**Review by JohelEGP — APPROVED** [review]

**Pull request merged**